backend/config.go

```diff
@@ -122,3 +123,26 @@ func (c *GrafanaCfg) AppURL() (string, error) {
 	}
 	return url, nil
 }
+
+type userAgentKey struct{}
+
+// UserAgentFromContext returns user agent from context.
+func UserAgentFromContext(ctx context.Context) (*useragent.UserAgent, error) {
```

Comment:
Nit: the returned `error` is always `nil`, I would probably change the signature to:

```suggestion
func UserAgentFromContext(ctx context.Context) *useragent.UserAgent {
```

Reply:
That gives me a compilation error:

```
too many return values
	have (*useragent.UserAgent, error)
	want (*useragent.UserAgent)
```

Reply:
I think I resolved this in 0f0d607. Please let me know if you meant something different!